### PR TITLE
feat: add clip endpoints

### PR DESCRIPTION
**Pull request by ahmedrangel**
Hello, I really liked your library, especially because it handles authentication very well. I thought it would be nice to add support for clip endpoints.

This PR aims to add a new endpoint for clips to which I have added two methods `getClip(clipId)` and `downloadClip(clipId)`

# getClip(clipId)
This method retrieves the details of a clip.
- No authentication required

Example of use
```js
// ... Kient
const clip_info = await client.api.clip.getClip('clip_01H811MXG4FBR62FXPE1AXABDH')
console.log(clip_info)
```
Output example
```json
{
  "clip": {
    "id": "clip_01HKXS6G6Y8R408ZZ6GJ4MCCE0",
    "livestream_id": "21657476",
    "category_id": "15",
    "channel_id": 1117152,
    "user_id": 17899999,
    "title": "Fly her out",
    "clip_url": "https://clips.kick.com/clips/dd/clip_01HKXS6G6Y8R408ZZ6GJ4MCCE0/playlist.m3u8",
    "thumbnail_url": "https://clips.kick.com/clips/dd/clip_01HKXS6G6Y8R408ZZ6GJ4MCCE0/thumbnail.png",
    "privacy": "CLIP_PRIVACY_PUBLIC",
    "likes": 3,
    "liked": false,
    "views": 4233,
    "duration": 30,
    "started_at": "2024-01-12T02:48:56.846Z",
    "created_at": "2024-01-12T02:50:08.426782Z",
    "is_mature": false,
    "video_url": "https://clips.kick.com/clips/dd/clip_01HKXS6G6Y8R408ZZ6GJ4MCCE0/playlist.m3u8",
    "view_count": 4233,
    "likes_count": 3,
    "category": {
      "id": 15,
      "name": "Just Chatting",
      "slug": "just-chatting",
      "responsive": "https://files.kick.com/images/subcategories/15/banner/b697a8a3-62db-4779-aa76-e4e47662af97",
      "banner": "https://files.kick.com/images/subcategories/15/banner/b697a8a3-62db-4779-aa76-e4e47662af97",
      "parent_category": "irl"
    },
    "creator": {
      "id": 17899999,
      "username": "ATOM313",
      "slug": "atom313",
      "profile_picture": "https://files.kick.com/images/user/17899999/profile_image/conversion/24969f0a-08a1-4549-ab0f-1096cc48bfee-thumb.webp"
    },
    "channel": {
      "id": 1117152,
      "username": "demisux",
      "slug": "demisux",
      "profile_picture": "https://files.kick.com/images/user/1157280/profile_image/conversion/8dc2d044-258a-41a9-92f7-abec7772d701-thumb.webp"
    }
  }
}
```

# downloadClip(clipId)
This method triggers the Kick API to download a clip and generates a temporary video/mp4 URL.
- Authentication required

Example of use
```js
// ... Kient
const clip_download = await client.api.clip.downloadClip('clip_01H811MXG4FBR62FXPE1AXABDH')
console.log(clip_download)
```

Output example
```json
{
  "url": "https://clips.kick.com/tmp/clip_01H811MXG4FBR62FXPE1AXABDH.mp4"
}
```

### CODE REVIEW

File: src/endpoints/clip/clip.endpoint.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { cast } from '@deepkit/type'
+import { BaseEndpoint } from '../endpoint.base'
+import type { DownloadClipResponse } from './dto/download-clip.response'
+import type { GetClipResponse } from './dto/get-clip.response'
```
Comment (zSoulweaver) [inline src/endpoints/clip/clip.endpoint.ts:4]: The `import type` statement unfortunately doesn't work with `@deepkit/type` as it needs the type information at runtime. We'll need to drop the `type` statement from the import and disable eslint for the follow line as it'll complain as it's only used in the generic type parameter for `cast`.

See [channel.instance.ts](https://github.com/zSoulweaver/kient/blob/master/src/endpoints/channel/instance/channel.instance.ts) for an example here.

File: src/endpoints/clip/clip.endpoint.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { KientApiError } from '@/errors'
+
+/**
+ * @category Endpoints
+ */
+
```
Comment (zSoulweaver) [inline src/endpoints/clip/clip.endpoint.ts:10]: nitpick: remove empty line

**Review by zSoulweaver — CHANGES_REQUESTED** [review]
Amazing! Great work implementing this and for putting some work into Kient :)

Few little things throughout before I merge it. I would also like to keep the library consistent in returning "instances" for every action that returns some data, as seen with the rest of the lib. You can look into [channel.endpoint.ts](https://github.com/zSoulweaver/kient/blob/master/src/endpoints/channel/channel.endpoint.ts) for an example of how that works, or let me know if you're struggling to understand anything.

I'd also like to know, as somebody who has contributed the first major addition to the library, how was your experience in adding to Kient and understanding the existing code base? I think it needs a bit of a refreshment in areas, but I'd be keen to hear any feedback.

**Comment by ahmedrangel** [comment]
Thank you for the correction! I'm not very experienced in Typescript but for this addition, I was able to understand it well by taking the time to analyze the code of what you did in the other endpoints.

**Pull request merged**